File: clients/go/examples/collections-find-and-rerank-sort-vector-vectorize.go
package main

import (
	"context"
	"fmt"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/filter"
	"github.com/datastax/astra-db-go/v2/astra/options"
	"github.com/datastax/astra-db-go/v2/astra/sort"
)

func main() {
	ctx := context.Background()

	// Get an existing collection
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	collection := database.Collection("**COLLECTION_NAME**")

	// Find documents
	cursor := collection.FindAndRerank(
		filter.F{},
		options.CollectionFindAndRerank().
			SetSort(sort.Hybrid("A tree in the woods")).
			SetIncludeSortVector(true),
	)

	// Inspect the sort vector
	vector := cursor.GetSortVector(ctx)
	fmt.Println(vector)
}

File: clients/java/examples/collections-find-and-rerank-sort-vector-vectorize.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.cursor.CollectionFindAndRerankCursor;
import com.datastax.astra.client.collections.commands.options.CollectionFindAndRerankOptions;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Sort;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Find documents
    CollectionFindAndRerankCursor<Document, Document> cursor =
        collection.findAndRerank(
            new CollectionFindAndRerankOptions()
                .sort(Sort.hybrid("A tree in the woods"))
                .includeSortVector(true));

    // Get the sort vector
    cursor
        .getSortVector()
        .ifPresent(
            vector -> {
              System.out.println(vector);
            });
  }
}
